lib/manageiq/appliance_console/database_admin.rb

```diff
@@ -88,29 +89,56 @@ def ask_smb_file_options
       end
 
       def ask_to_delete_backup_after_restore
-        if backup_type == LOCAL_FILE
+        if action == :restore && backup_type == LOCAL_FILE
           say("The local database restore file is located at: '#{uri}'.\n")
           @delete_agree = agree("Should this file be deleted after completing the restore? (Y/N): ")
         end
       end
 
       def confirm_and_execute
-        say("\nNote: A database restore cannot be undone.  The restore will use the file: #{uri}.\n")
-        if agree("Are you sure you would like to restore the database? (Y/N): ")
-          say("\nRestoring the database...")
-          rake_success = ManageIQ::ApplianceConsole::Utilities.rake(task, task_params)
-          if rake_success && delete_agree
-            say("\nRemoving the database restore file #{uri}...")
-            File.delete(uri)
-          elsif !rake_success
-            say("\nDatabase restore failed. Check the logs for more information")
-          end
+        if allowed_to_execute?
+          processing_message
+          run_rake
         end
         press_any_key
       end
 
+      def allowed_to_execute?
+        if action == :restore
+          say("\nNote: A database restore cannot be undone.  The restore will use the file: #{uri}.\n")
+          agree("Are you sure you would like to restore the database? (Y/N): ")
+        else
+          true
+        end
+      end
+
+      def file_menu_args
+        [
+          action == :restore ? "Restore Database File" : "Backup File Location",
```

Comment:
Reading through this class I've seen the file to be used for a restore called the "backup file" a few times so this was a bit confusing.
Would something like "Backup Result File Name" make more sense here? (not sold on the name, looking for suggestions ... @jrafanie don't you love naming things?! :wink: )

Reply:
I mean, I am not sold on anything myself, so if you would prefer that, I can go with that option as well.

Reply:
@carbonin Would `"Backup Output File Name"` be better?

Reply:
Yeah, that sounds good :+1: